pkg/aws/s3/objects.go

```diff
@@ -192,3 +193,22 @@ func ListObjectsWithPrefix(s3Session *s3.S3, r *ListObjectsReq) ([]*s3.Object, e
 
 	return res.Contents, nil
 }
+
```

Comment:
Could you add a detailed function definition, as provided in the other functions in same file.
We are maintaining it for easier comprehension by other developers as this is a central library.